cli/submit.go

```diff
@@ -1,96 +1,170 @@
-    Args:  &SubmitArgs{},
-    Run:   SubmitRun,
+	Name:  "submit",
+	Short: "Submit your Keyset to a git repository.",
+	Args:  &SubmitArgs{},
+	Run:   SubmitRun,
 }
 
 type SubmitArgs struct {
-    Args []string
+	Args []string
 }
 
+//A place to put special repo "aliases"
+var SpecialRepos = map[string]string{
+	"core": "https://github.com/arkenproject/core-keyset.git",
+}
+
+//SubmitRun generates a keyset file and then clones the Github repo at the given
+//url, adds the keyset file, commits it, and pushes it, and then deletes the repo
+//once everything is done or if anything goes wrong before completion. With all
+//of those steps, there are MANY possible points of failure. If anything goes
+//wrong, the error will be log.Fatal'd and the repo will we deleted from
+//its temporary location at .ait/sources. Users are not meant to deal with the
+//repos directly at any point so it and the keyset file are basically ephemeral
+//and only exist on disk while this command is running.
 func SubmitRun(_ *cmd.RootCMD, c *cmd.CMD) {
-    args := c.Args.(*SubmitArgs).Args
-    if len(args) < 2 {
-        log.Fatal("Not enough arguments, expected repository name and path to Keyset")
-    }
-    repoName, keysetPath := args[0], args[1]
-    repo := add(repoName, keysetPath)
-    commit(repo)
-    push(repo)
+	args := c.Args.(*SubmitArgs).Args
+	if len(args) < 1 {
+		log.Fatal("Not enough arguments, expected repository url")
+	}
+	if s, _ := utils.GetFileSize(utils.AddedFilesPath); s == 0 {
+		log.Fatal(`No files are currently added, nothing to submit. Use
+    ait add <files>...
+to add files for submission.`)
+	}
+	url, contains := SpecialRepos[args[0]]
+	if !contains {
+		url = args[0]
+	}
+	target := filepath.Join(".ait", "sources", utils.GetRepoName(url))
+	if !utils.FileExists(target) {
+		keysets.Clone(url)
+	}
+	repo, err := git.PlainOpen(target)
+	if err != nil {
+		cleanup(target)
+		log.Fatal(err)
+	}
+	keysetPath := "test.ks"
+	err = keysets.Generate(filepath.Join(target, keysetPath))
+	if err != nil {
+		cleanup(target)
+		log.Fatal(err)
+	}
+	add(repo, keysetPath, target)
+	commit(repo, target)
+	push(repo, target)
+	cleanup(target)
+	//somewhere in here I'm going to have to add pull request support
+}
+
+//add adds the keyset file at the given path to the repo.
+//Effectively: git add keysetPath
+func add(repo *git.Repository, keysetPath, repoPath string) {
+	tree, err := repo.Worktree()
+	if err != nil {
+		cleanup(repoPath)
+		log.Fatal(err)
+	}
+	_, err = tree.Add(keysetPath)
+	if err != nil {
+		cleanup(repoPath)
+		log.Fatal(err)
+	}
+}
+
+//commit attempts to commit the file that was previously added.
+func commit(repo *git.Repository, repoPath string) {
+	tree, err := repo.Worktree()
+	if err != nil {
+		cleanup(repoPath)
+		log.Fatal(err)
+	}
+	msg := CollectCommit()
+	opt := &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  "name", //get from config whenever that's ready
+			Email: "someone@somehwere.com",
```

Comment:
This should be available at `config.Global.Git.Name` and `config.Global.Git.Email` as of this weekend.

Reply:
just added, you'll see that soon